crates/api/src/peer_store.rs

```diff
@@ -0,0 +1,62 @@
+//! Peer-store related types.
+
+use crate::*;
+use std::sync::Arc;
+
+/// Represents the ability to store and query agents.
+pub trait PeerStore: 'static + Send + Sync + std::fmt::Debug {
+    /// Insert agents into the store.
+    fn insert(
+        &self,
+        agent_list: Vec<Arc<agent::AgentInfoSigned>>,
+    ) -> BoxFut<'_, K2Result<()>>;
+
+    /// Get an agent from the store.
+    fn get(
+        &self,
+        agent: id::AgentId,
+    ) -> BoxFut<'_, K2Result<Option<Arc<agent::AgentInfoSigned>>>>;
+
+    /// Get all agents from the store.
+    fn get_all(&self)
+        -> BoxFut<'_, K2Result<Vec<Arc<agent::AgentInfoSigned>>>>;
+
+    /// Get the complete list of agents we know about that
+    /// claim storage_arcs that overlap the provided storage arc.
+    /// This function, provided with the storage arc of a currently active
+    /// local agent, will return a list of agents with whom
+    /// to gossip (and also the local agent). If there are multiple
+    /// local agents in this space, you'll need to call this function
+    /// multiple times and union the results.
+    fn get_overlapping_storage_arc(
+        &self,
+        arc: agent::BasicArc,
+    ) -> BoxFut<'_, K2Result<Vec<Arc<agent::AgentInfoSigned>>>>;
+
+    /// Get a list of agents sorted by nearness to a target basis location.
+    /// If the
```

Comment:
Missing something in this comment?

Reply:
Done: https://github.com/holochain/kitsune2/pull/19/commits/a368c14881972682fbaf135d2bc702a8a41758b2